src/main/java/io/anserini/document/JsonDocument.java

```diff
@@ -34,45 +36,38 @@
 public class JsonDocument implements SourceDocument {
   protected String id;
   protected String contents;
-  private JsonElement raw;
+  private ArrayNode node;
   private int i;
 
-  public JsonDocument(String path) throws FileNotFoundException {
-    JsonParser parser = new JsonParser();
+  public JsonDocument(String path) {
     try {
-      raw = parser.parse(new BufferedReader(new FileReader(path))).getAsJsonArray();
-    } catch (IllegalStateException e1) {
-      try {
-        raw = parser.parse(new BufferedReader(new FileReader(path))).getAsJsonObject();
-      } catch (IllegalStateException e2) {
-        raw = parser.parse(new BufferedReader(new FileReader(path))).getAsJsonNull();
-      }
+      JsonParser jsonParser = new JsonFactory().createParser(new BufferedReader(new FileReader(path)));
+      ObjectMapper objectMapper = new ObjectMapper();
+      objectMapper.configure(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY, true);
+      node = objectMapper.readTree(jsonParser);
+    } catch (IOException e) {
+      node = null;
```

Comment:
My understanding is that this block is actually a logic if-else block instead of try-catch block?
If so, we should add comments to not confuse everyone.

Reply:
Yes, will add comments now.